tests/integration/api/test_source.py

```diff
@@ -0,0 +1,223 @@
+
+    sources = models.Source.objects.bulk_create(
+        [
+            models.Source(
+                uuid=uuid.uuid4(),
+                name="test-source-1",
+                type="ansible.eda.range",
+                args={"limit": 5, "delay": 1},
+                activation=activation,
+                user=activation.user,
+                decision_environment=init_db.decision_environment,
+            ),
+            models.Source(
+                uuid=uuid.uuid4(),
+                name="test-source-2",
+                type="ansible.eda.range",
+                args={"limit": 6, "delay": 2},
+                activation=activation,
+                user=activation.user,
+                decision_environment=init_db.decision_environment,
+            ),
+        ]
+    )
+
+    response = client.get(f"{api_url_v1}/sources/")
+    assert response.status_code == status.HTTP_200_OK
+    assert len(response.data["results"]) == 2
+    response.data["results"][0]["uuid"] = sources[0].uuid
+    response.data["results"][0]["type"] = sources[0].type
+    response.data["results"][0]["name"] = sources[0].name
+    response.data["results"][0]["args"] = sources[0].args
+
+    check_permission_mock.assert_called_once_with(
+        mock.ANY, mock.ANY, ResourceType.SOURCE, Action.READ
+    )
+
+
+@pytest.mark.django_db
+def test_retrieve_source(
+    client: APIClient, init_db, check_permission_mock: mock.Mock
+):
+    activation = init_db.activation
+
+    source = models.Source.objects.create(
+        uuid=uuid.uuid4(),
+        name="test-source-1",
+        type="ansible.eda.range",
+        args={"limit": 5, "delay": 1},
+        activation=activation,
+        user=activation.user,
+        decision_environment=init_db.decision_environment,
+    )
+
+    response = client.get(f"{api_url_v1}/sources/{source.uuid}/")
+    assert response.status_code == status.HTTP_200_OK
+    assert response.data["name"] == source.name
+    assert response.data["type"] == source.type
+
+    check_permission_mock.assert_called_once_with(
+        mock.ANY, mock.ANY, ResourceType.SOURCE, Action.READ
+    )
+
+
+@pytest.mark.django_db
+def test_create_source(
+    client: APIClient, init_db, check_permission_mock: mock.Mock
+):
+    data_in = {
+        "name": "test_source",
+        "type": "ansible.eda.generic",
+        "args": {"limit": 1, "delay": 5},
+        "activation_id": init_db.activation.id,
+    }
+    response = client.post(f"{api_url_v1}/sources/", data=data_in)
+    assert response.status_code == status.HTTP_201_CREATED
+    result = response.data
+    assert result["name"] == "test_source"
+    assert result["type"] == "ansible.eda.generic"
+    assert result["args"] == "delay: 5\nlimit: 1"
+    assert result["user"] == "test.admin"
+
+    check_permission_mock.assert_called_once_with(
+        mock.ANY, mock.ANY, ResourceType.SOURCE, Action.CREATE
+    )
+
+
+@pytest.fixture
+def init_db():
+    test_rulesets_sample = """
```

Comment:
@hsong-rh Is this needed if we are not creating an activation before creating the source. The Activation is post create